### PR TITLE
[@mantine/form] add asyncResolver on use-form

### PR DESCRIPTION
**Pull request by auronsan1st**
https://github.com/mantinedev/mantine/issues/1100

to check promise need call function ex schema(null) I think there will be performance loss on other resolver ex: yup, zoi. so I decided just use asyncResolver props.

my suggest to make not bigger code should split between async and sync so no hidden bug will occur.
the schema types depend on many place. so I can't change to use Promise<schema>. for now I just bypass it by any when init useForm

### CODE REVIEW

File: src/mantine-form/src/resolvers/async-test-resolver.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { renderHook, act } from '@testing-library/react-hooks';
```
Comment (rtivital) [inline src/mantine-form/src/resolvers/async-test-resolver.ts:1]: All tests files should be in tests folder.

File: src/mantine-demos/src/demos/form/Form.demo.zod.tsx [hunk 2]
```diff
@@ -66,7 +66,8 @@ const schema = z.object({
 
 function Demo() {
   const form = useForm({
-    schema: zodResolver(schema),
+    asyncResolver: true,
+    schema: zodResolver(schema) as any,
```
Comment (rtivital) [inline src/mantine-demos/src/demos/form/Form.demo.zod.tsx:70]: API does not fell nice. Can you make it as before `schema: zodResolver(schema)`, without `asyncResolver` field and `as any`?

**Review by rtivital — COMMENTED** [review]
Currently, I do not really like what I'm seeing, async api feels very strange and unintuitive.

File: src/mantine-form/src/resolvers/async-test-resolver.ts (hunk 1 above)
Comment (auronsan) [reply src/mantine-form/src/resolvers/async-test-resolver.ts:1]: it's test helper async version of `src/mantine-form/src/resolvers/test-resolver.ts`

**Comment by auronsan** [comment]
> Currently, I do not really like what I'm seeing, async api feels very strange and unintuitive.

yeah I feel same. but if can need support it right. actually if https://github.com/mantinedev/mantine/issues/1100
they can check theirselves and throw error when trigger on submit

**Comment by rtivital** [comment]
Alright, I think this issue will require more investigation before it can be merged, getting the api right for these use cases seems to be tricky

**Pull request closed without merging**